### ISSUE TITLE
Add ClassLoader to CompileOptions in order to allow for overriding the ClassLoader

### ISSUE THREAD

**Issue by vksiva** [jOOQ/jOOR#72]
### Expected behavior and actual behavior:
Using the example as presented in the README, 
```
Supplier<String> supplier = Reflect.compile(
    "com.example.HelloWorld",
    "package com.example;\n" +
    "class HelloWorld implements java.util.function.Supplier<String> {\n" +
    "    public String get() {\n" +
    "        return \"Hello World!\";\n" +
    "    }\n" +
    "}\n").create().get();

// Prints "Hello World!"
System.out.println(supplier.get());
```
If the text has to be changed to "Hi World!" and the class is recompiled, the below code still prints "Hello World!"
```
 supplier = Reflect.compile(
    "com.example.HelloWorld",
    "package com.example;\n" +
    "class HelloWorld implements java.util.function.Supplier<String> {\n" +
    "    public String get() {\n" +
    "        return \"Hi World!\";\n" +
    "    }\n" +
    "}\n").create().get();

// Still prints "Hello World!"
System.out.println(supplier.get());
```
One way to ensure the class gets recompiled is by using different classloaders during compile. Can we use a custom classloader for the Reflect.compile() method. It was raised in #64.

### Steps to reproduce the problem:
Mentioned above.

### Versions:

- jOOR: 0.9.9
- Java: 1.8

**Comment by lukaseder** [comment]
Thanks for reporting this. Indeed, we've solved this in the next release through https://github.com/jOOQ/jOOR/issues/67

**Comment by lukaseder** [comment]
In fact, no. `CompileOptions` does not allow for overriding the `ClassLoader` yet. Will implement this soon.

**Comment by lukaseder** [comment]
The custom class loader feature has been implemented. But I will also implement a breaking change that avoids re-loading a previously compiled class by default. I don't think that's what anyone really wants. See #76

**Comment by lukaseder** [comment]
Hmm, no in fact #76 cannot be implemented. The current implementation prevents `LinkageError` in some cases.

**Comment by lukaseder** [comment]
> Hmm, no in fact #76 cannot be implemented. The current implementation prevents `LinkageError` in some cases.

I'm no longer running into these errors from: 
https://github.com/jOOQ/jOOR/issues/72#issuecomment-444077568

Perhaps that was a JDK bug, at the time?

**Comment by lukaseder** [comment]
Fixed for jOOR 0.9.15

**Comment by lukaseder** [comment]
Time for a release, too.
